### ISSUE TITLE
Links in GitGraph ?

### ISSUE THREAD

**Issue by cpivot** [mermaid-js/mermaid#5599]
### Proposal

Hello
I'm looking to use links or buttons in GitGhrap, especially in commit ids.
Is this possible?
Thanks

### Example

gitGraph TB:
        commit id:"<a href='GitHub.com'>GitHub</a>"

**Comment by jasonatcb** [comment]
I'd love this feature as well.

**Comment by ivaigult** [comment]
I’d love to see hyperlink support in gitGraph, and wanted to describe a concrete use-case.

I often use Mermaid’s gitGraph diagrams to visualize how a feature landed across multiple branches — for example, to show when a change was merged into main, then cherry-picked into a release branch, or rebased onto another integration line.

It would be incredibly useful if I could add a hyperlink to each commit (or branch) node so that clicking it would open the corresponding PR, Gerrit change, or commit in my code review system.

For example, something like:
```
gitGraph
  commit id: "1234" tag: "feature-X" href: "https://github.com/org/repo/pull/1234"
  branch release
  checkout release
  merge feature-X href: "https://github.com/org/repo/pull/5678"
```

This would make Mermaid diagrams a powerful bridge between documentation and actual code review context — helping devs and reviewers quickly trace how a feature propagated across branches.

Even a simple href: attribute or HTML anchor tag in the commit label would cover most use-cases.

I understand that gitGraph is still evolving, but since it’s often used for visualizing real Git history, the ability to connect nodes to real commits or PRs would be a major usability boost.

Thanks for considering this!

**Comment by mkobit** [comment]
I'm interested in tackling this and have been exploring the implementation. Before getting too much further, I'd like to confirm the desired syntax.

### Clickable Elements

`gitGraph `has several elements that could potentially support click handlers:

| Element | Has Identifier | Notes |
|---------|----------------|-------|
| Commit | Optional `id:` | Main use case |
| Merge | Optional `id:` | Creates a commit, so similar to commit |
| Cherry-pick | Optional `id:` | Also creates a commit |
| Tag | Name string | Attached to a commit; a commit can have multiple tags |
| Branch | Name | The branch label itself, not the commits on it |

### Syntax Sketch

Here's what the various click handlers might look like:

```
gitGraph
    commit id: "init"
    branch feature
    commit id: "feat-1"
    commit id: "feat-2" tag: "v1.0"
    checkout main
    merge feature id: "merge-1"
    
    %% Commits (includes merges and cherry-picks with IDs)
    click "init" "https://github.com/repo/commit/abc"
    click "feat-1" "https://github.com/repo/commit/def"
    click "merge-1" "https://github.com/repo/pull/42"
    
    %% Branch labels
    click branch feature "https://github.com/repo/tree/feature"
    
    %% Tags
    click tag "v1.0" "https://github.com/repo/releases/tag/v1.0"
```

###  Decisions

**1. Commits/Merges/Cherry-picks:** These all support `id:`, so a single `click "id" "url"` pattern (matching flowchart) would cover all three. This seems straightforward depending on the overall desired syntax, and if a keyword `commit` would be used as well (see below suggestions).

**2. Branches:** The branch name could collide with a commit ID. Options:
   - Require keyword: `click branch <name> "url"` (explicit, no ambiguity)
   - Assume commit IDs take precedence (implicit, could be confusing)
   - Allow specifying the `click` handler on the `branch` declaration. This couples declaration to the styling.

**3. Tags:** Similar ambiguity question. Also, multiple tags can exist on one commit
   - `click tag "<name>" "url"` (explicit)
   - Allow specifying the `click` handler on the `tag` declaration. This couples declaration to the styling, and is the same as above.

**4. Tooltips and targets:** Should these match flowchart's full syntax?
   - `click "id" "url" "tooltip" _blank`

**What I'm looking for:** Confirmation on scope, and ideally an example of the desired final syntax so I can implement against a clear target.

**Comment by mkobit** [comment]
I opened https://github.com/mermaid-js/mermaid/pull/7292 for this as I think working code is easier to discuss than hypotheticals.

**Comment by mkobit** [comment]
I'm still waiting on feedback. I'm not planning to keep the branch up to date with merge conflicts until there is more momentum and support for getting it merged, as it is currently unreviewed after nearly 3 months.

**Comment by ashishjain0512** [comment]
@mkobit Great initiative and progress. Yes, my initial hunch was to stay as close as possible to the flowchart syntax. Going through your decision description, it's in line with our expectations.

We are bringing back traction in the project, and we will review your PR soon (maybe resolve the conflicts and sync it to the latest develop)

**Comment by syh7** [comment]
Any update on this @ashishjain0512 ?
